src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -790,6 +790,20 @@ fn is_in_follow(tok: &quoted::TokenTree, frag: &str) -> Result<bool, (String, &'
                 // harmless
                 Ok(true)
             },
+            "vis" => {
+                // Explicitly disallow `priv`, on the off chance it comes back.
+                match *tok {
+                    TokenTree::Token(_, ref tok) => match *tok {
+                        Comma => Ok(true),
+                        ModSep => Ok(true),
```

Comment:
>I allowed ModSep because you can already have `pub ::path::to::a_type`

This seems quite inconsistent, you can already have `pub ARBITRARY_TYPE`, but only a subset of paths is allowed.

This should probably use all `can_begin_type` tokens (in addition to comma) + `ty/path/ident` for `MetaVarDecl`s.
Or all `is_path_start` tokens (in addition to comma) + `ty/path/ident` for `MetaVarDecl`s if we want to be conservative.